src/data/finset/basic.lean

```diff
@@ -1682,6 +1696,19 @@ begin
       assume x hx, match x, finset.mem_image.1 hx with _, ⟨a, ha, rfl⟩ := hf a ha end
 end
 
+/--
+If there are more pigeons than pigeonholes, then there are two pigeons
+in the same pigeonhole.
+-/
+lemma pigeonhole {s : finset α} {t : finset β} (hc : t.card < s.card)
```

Comment:
`mathlib` generally avoids declaration names like `pigeonhole`. It's better to follow the standard naming convention and make sure that "Pigeonhole principle" appears at the beginning of the doc string. As for what the mathlib name of this should be, I'm not sure, maybe better to get some opinions here.

Reply:
Yeah, I wasn't sure what the mathlib name should be, so I left it with the common name as a placeholder.  I think for a little bit I had `not_inj_if_card_lt`.